core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -136,12 +136,6 @@ private[spark] class AppStatusStore(
     store.read(classOf[StageDataWrapper], Array(stageId, stageAttemptId)).locality
   }
 
-  // SPARK-26119: we only want to consider successful tasks when calculating the metrics summary,
```

Comment:
What about this comment though - is this not still an issue?

Reply:
Thanks @srowen.I think the behavior will be same for live UI after this PR. Only successful tasks' metrics will be considered for summary calculation. Only behavior change is for disk store case. We added UTs in AppStatusStoreSuite

Reply:
I mean the comment below it, where it says this is very expensive, etc. This seems to use the 'slow' code path in all cases?

Reply:
I think "very expensive" meant for disk store case. Because of the deserialising overhead, the InMemory code path for disk store is expensive. Vanzin has commented in the previous PR https://github.com/apache/spark/pull/23088#discussion_r237197020

Reply:
Also I checked performance analysis with 1 lac tasks

1. bin/spark-shell 
`sc.parallelize(1 to 100000, 100000).count()`

Time taking to load the stage page:
1. InMemory store (both live and History): ~ 8-9 sec (before and after the PR)
2. DiskStore  -> ~9 sec (without PR but incorrect results if non successful tasks)
                          ~14 sec (with PR but correct results)

Reply:
And these loading time can be further reduced. Because currently stage UI requires only 9 tasks metrics summary, I think. But we are calculating summaries of 28 tasks metrics. Instead if we compute only 9 for UI, again the loading time can be reduced. I'll raise a new PR for that optimisation too. cc @vanzin